pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTStep.java

```diff
@@ -94,7 +94,20 @@ public String toGroovy() {
             }
         }
 
-        return name + "(" + (argList != null ? argList.toGroovy() : "") + ")";
+        return withOrWithoutParens(argList);
+    }
+
+    private String withOrWithoutParens(ModelASTArgumentList argList) {
+        if (argList == null) {
+            return name + "()";
+        } else if (!(this instanceof ModelASTTreeStep) &&
+                argList instanceof ModelASTSingleArgument &&
+                // Special-casing for list/map args since they still need parentheses.
+                !argList.toGroovy().startsWith("[")) {
```

Comment:
_nit_ could potentially be better optimized by doing just one call to `toGroovy()`